File: sdks/thunkmetrc-py/wrapper/src/thunkmetrc/wrapper/services/waste_methods_service.py
from typing import Any, Optional, List, AsyncGenerator
from thunkmetrc.client import MetrcClient
from ..ratelimiter import MetrcRateLimiter
from ..models import *
from ..utils import iterate_pages, deserialize_page

class WasteMethodsService:
    def __init__(self, client: MetrcClient, limiter: MetrcRateLimiter):
        self.client = client
        self.limiter = limiter

    async def get_waste_methods_waste_methods(self, body: Any = None) -> List['WasteMethod']:
        """
        Retrieves all available waste methods.
        GET /wastemethods/v2
        Parameters:
        """
        async def op():
            return await self.client.waste_methods.get_waste_methods_waste_methods(body=body,
            )
        return await self.limiter.execute(None, True, op)

File: sdks/thunkmetrc-ts/wrapper/src/services/WasteMethodsService.ts

import { MetrcClient as InternalMetrcClient } from '@thunkier/thunkmetrc-client';
import { MetrcRateLimiter } from '../RateLimiter';
import { iteratePages } from '../Utils';
import { 
    PaginatedResponse,
    WasteMethod,
} from '../models';

export class WasteMethodsService {
  constructor(private client: InternalMetrcClient, private rateLimiter: MetrcRateLimiter) {}

  /**
   * Retrieves all available waste methods.
   * GET GetWasteMethodsWasteMethods
   */
  public async getWasteMethodsWasteMethods(body?: any): Promise<WasteMethod[]> {
    return this.rateLimiter.execute(null,true,
        () => this.client.getWasteMethodsWasteMethods(body,
        )
    );
  }
}

